### ISSUE TITLE
Parameter handling in NetworkAPI

### ISSUE THREAD

**Issue by dkeeney** [htm-community/htm.core#842]
Parameter handling (the getters and setters) consume a lot of code in NetworkAPI.  The question is if it is worth the effort to re-engineer this logic ... and probably break the API.

Is this over-kill?

**Comment by breznak** [comment]
you're right...the parameter handling is always a chunk of code! On the other hand, once done for each region, we don't hear about it (and it's not any pain when USING the code), so I'm not sure if worth extra work/breaking. 

That being said, if you have an idea or PoC, I'd gladly discuss that :) 
My same wishlist is for Regions (and even classes) to accept a JSON as a param config. That would enable users "tuning params" without need to recompile or change the code. 

We shall start collecting ideas for breaking the (N)API and v2.

**Comment by dkeeney** [comment]
>  My same wishlist is for Regions (and even classes) to accept a JSON as a param config. That would enable users "tuning params" without need to recompile or change the code.

We are almost there.  Basically that is what the REST api does.
